### PR TITLE
CEDS-1687 Make reference in transport details and border transport mandatory

### PR DESCRIPTION
**Pull request by bambuchaAdm**
This PR disallow empty reference. Some fixes in messages keys, some clean up in model.

There is new extracted base class for testing `FormSpec`. Have some matcher and scala test Type Classes which make testing of play forms much easier.

Include some content changes address around this issue.

### CODE REVIEW

**Comment by PatrykRudnicki** [comment]
Please fix test: `views.declaration.TransportDetailsViewSpec`

File: app/models/declaration/wcodeclarationModels.scala [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 package models.declaration
 
+//FIXME deadcode
 case class BorderTransportMeans(
```
Comment (PatrykRudnicki) [inline app/models/declaration/wcodeclarationModels.scala]: It's wcodecalrationModels, so probably it's unused. In this case please remove it

File: test/services/cache/ExportsDeclarationBuilder.scala [hunk 2]
```diff
@@ -159,7 +159,7 @@ trait ExportsDeclarationBuilder {
   def withBorderTransport(
     borderModeOfTransportCode: String = "",
     meansOfTransportOnDepartureType: String = "",
-    meansOfTransportOnDepartureIDNumber: Option[String] = None
+    meansOfTransportOnDepartureIDNumber: String = "1"
```
Comment (PatrykRudnicki) [inline test/services/cache/ExportsDeclarationBuilder.scala]: Why not empty string?

File: test/services/cache/ExportsDeclarationBuilder.scala [hunk 3]
```diff
@@ -227,7 +227,7 @@ trait ExportsDeclarationBuilder {
     meansOfTransportCrossingTheBorderNationality: Option[String] = None,
     container: Boolean = false,
     meansOfTransportCrossingTheBorderType: String = "",
-    meansOfTransportCrossingTheBorderIDNumber: Option[String] = None,
+    meansOfTransportCrossingTheBorderIDNumber: String = "1",
```
Comment (PatrykRudnicki) [inline test/services/cache/ExportsDeclarationBuilder.scala]: The same here

File: test/services/cache/ExportsDeclarationBuilder.scala (hunk 2 above)
Comment (bambuchaAdm) [reply test/services/cache/ExportsDeclarationBuilder.scala]: Count not be empty

File: test/services/cache/ExportsDeclarationBuilder.scala (hunk 3 above)
Comment (bambuchaAdm) [reply test/services/cache/ExportsDeclarationBuilder.scala]: Could not be empty - it is now mandatory field

File: test/services/cache/ExportsDeclarationBuilder.scala (hunk 2 above)
Comment (PatrykRudnicki) [reply test/services/cache/ExportsDeclarationBuilder.scala]: Yes, but `borderModeOfTransportCode` cannot be empty too and it contain empty string

**Pull request merged**